### PR TITLE
AMLII-864-E2E/journald log collection

### PR DESCRIPTION
**Pull request by DDuongNguyen**
#### What does this PR do?
This PR introduces a new end-to-end (e2e) test to verify the agent's capability to tail journald logs in Linux OS. This serves as a step in our effort to shift our quality assurance testing towards more comprehensive e2e testing.

#### Motivation
This PR purpose is to ensure robustness in the agent's log-tailing feature, especially in various edge cases such as:
- Tailing all logs from jouranld
- Using including units
- Using excluding units

By converting this QA test to an e2e test, we aim to catch potential issues early in the development cycle in an automated way.

### CODE REVIEW

File: test/new-e2e/tests/agent-metric-logs/log-agent/linux-log/journald/journald_tailing_test.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+// LinuxJournaldFakeintakeSuite defines a test suite for the log agent interacting with a virtual machine and fake intake.
+type LinuxJournaldFakeintakeSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+//go:embed log-config/journald.yaml
+var logConfig []byte
+
+//go:embed log-config/include.yaml
+var logConfig2 []byte
```
Comment (carlosroman) [inline test/new-e2e/tests/agent-metric-logs/log-agent/linux-log/journald/journald_tailing_test.go:36]: Just nit pick, but wonder if these configs could have a better name? Makes it easier to understand what the config is doing for that test.

```suggestion
var logConfigWithJustInclude []byte
```

Comment (DDuongNguyen) [reply test/new-e2e/tests/agent-metric-logs/log-agent/linux-log/journald/journald_tailing_test.go]: I changed it to:
- logBasicConfig
- logIncludeConfig
- logExcludeConfig

File: test/new-e2e/tests/agent-metric-logs/log-agent/linux-log/journald/journald_tailing_test.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+		if assert.Truef(c, agentReady, "Agent is not ready after restart") {
+			// Check that the datadog-agent.service log is not collected, specifically logs from the check runners
+			utils.CheckLogsNotExpected(s, "no-datadog", "running check")
+		}
+	}, 1*time.Minute, 5*time.Second)
+}
+
+// appendJournaldLog appends a log to journald.
+func appendJournaldLog(s *LinuxJournaldFakeintakeSuite, content string, recurrence int) {
+	t := s.T()
```
Comment (rayz) [inline test/new-e2e/tests/agent-metric-logs/log-agent/linux-log/journald/journald_tailing_test.go:172]: suggestion:
add `s.T().Helper()` to indicate that this is a helper

**Review by carlosroman — APPROVED** [review]

**Comment by DDuongNguyen** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable DEPLOY_AGENT=true
